btr-api/src/btr_api/services/auth.py

```diff
@@ -66,59 +72,129 @@ def init_app(self, app: Flask):
-                                data=data,
-                                headers={'content-type': 'application/x-www-form-urlencoded'},
-                                auth=(self.svc_acc_id, self.svc_acc_secret),
-                                timeout=self.sso_svc_timeout)
+            res = requests.post(
+                url=self.sso_svc_token_url,
+                data=data,
+                headers={'content-type': 'application/x-www-form-urlencoded'},
+                auth=(self.svc_acc_id, self.svc_acc_secret),
+                timeout=self.sso_svc_timeout,
+            )
             if res.status_code != HTTPStatus.OK:
                 raise ConnectionError({'statusCode': res.status_code, 'json': res.json()})
             return res.json().get('access_token')
         except exceptions.Timeout as err:
             self.app.logger.debug('SSO SVC connection timeout: %s', err.with_traceback(None))
-            raise ExternalServiceException(HTTPStatus.GATEWAY_TIMEOUT,
-                                           [{'message': 'Unable to get service account token.',
-                                             'reason': err.with_traceback(None)}]) from err
+            raise ExternalServiceException(
+                HTTPStatus.GATEWAY_TIMEOUT,
+                [{'message': 'Unable to get service account token.', 'reason': err.with_traceback(None)}],
+            ) from err
         except Exception as err:  # noqa: B902
             self.app.logger.debug('SSO SVC connection failure: %s', err.with_traceback(None))
-            raise ExternalServiceException(HTTPStatus.SERVICE_UNAVAILABLE,
-                                           [{'message': 'Unable to get service account token.',
-                                             'reason': err.with_traceback(None)}]) from err
+            raise ExternalServiceException(
+                HTTPStatus.SERVICE_UNAVAILABLE,
+                [{'message': 'Unable to get service account token.', 'reason': err.with_traceback(None)}],
+            ) from err
 
     def get_authorization_header(self, request) -> str:
         """Gets authorization header from request."""
         authorization_header = request.headers.get('Authorization', None)
         if not authorization_header:
-            error = f'Missing authorization header: {request.headers}'
+            error = f"Missing authorization header: {request.headers}"
             self.app.logger.debug('Cannot find authorization_header in request.')
             raise AuthException(error=error, status_code=HTTPStatus.UNAUTHORIZED)
 
         return authorization_header
 
+    # TODO: Add caching
+    def product_authorizations(self, request, account_id: str) -> bool:
+        """Authorize the user for access to the service."""
+        if not account_id:
+            error = 'Missing account_id'
+            self.app.logger.debug(error)
+            request.product_resp = {}
+            return True
```

Comment:
is it authorized if there is no account_id ?
What does return value do ?

Reply:
The comment is perhaps misleading as the function doesn't restrict access it just sets the products I'll update the comment